packages/block-library/src/pullquote/deprecated.js

```diff
@@ -152,24 +273,60 @@ const deprecated = [
 				</figure>
 			);
 		},
-		migrate( { className, figureStyle, mainColor, ...attributes } ) {
+		migrate( {
+			className,
+			figureStyle,
+			mainColor,
+			customMainColor,
+			customTextColor,
+			...attributes
+		} ) {
 			const isSolidColorStyle = includes( className, SOLID_COLOR_CLASS );
+			const style = {};
+			if ( customMainColor ) {
+				if ( ! isSolidColorStyle ) {
+					// Block supports: Set style.border.color if a deprecated block has a default style and a `customMainColor` attribute.
+					style.border = {
+						color: customMainColor,
+					};
+				} else {
+					// Block supports: Set style.color.background if a deprecated block has a solid style and a `customMainColor` attribute.
+					style.color = {
+						background: customMainColor,
+					};
+				}
+			}
+
+			// Block supports: Set style.color.text if a deprecated block has a `customTextColor` attribute.
+			if ( customTextColor ) {
+				style.color = {
+					...style?.color,
+					text: customTextColor,
+				};
+			}
 			// If is the default style, and a main color is set,
-			// migrate the main color value into a custom color.
-			// The custom color value is retrived by parsing the figure styles.
+			// migrate the main color value into a custom border color.
+			// The custom border color value is retrieved by parsing the figure styles.
 			if ( ! isSolidColorStyle && mainColor && figureStyle ) {
 				const borderColor = parseBorderColor( figureStyle );
 				if ( borderColor ) {
 					return {
 						...attributes,
 						className,
-						customMainColor: borderColor,
+						// Block supports: Set style.border.color if a deprecated block has `mainColor`, inline border CSS and is not a solid color style.
+						style: {
+							border: {
+								color: borderColor,
+							},
+						},
 					};
 				}
 			}
 			return {
 				className,
-				mainColor,
+				backgroundColor: isSolidColorStyle ? mainColor : undefined,
+				borderColor: isSolidColorStyle ? undefined : mainColor,
+				style,
```

Comment:
This will currently add `style`, even if it's an empty object, see https://github.com/WordPress/gutenberg/pull/30951/files#diff-cc59bbe727811591bfc343d05c2f46e36e9562c3e01319de8044160d437a8db5R1. We could consider omitting it in that case.

Reply:
Nice, thanks for catching that one 😄